### PR TITLE
fix: upgrade deno_ast to 0.23

### LINKED ISSUE #8529: Parse errors from SWC could be better
Currently if SWC parser encounters error its output is pretty minimal:

```ts
import b, { c } from "./b.js";

console.log(b);
console.log(c
```

```
error: Unexpected eof at file:///Users/biwanczuk/dev/deno/a.js:4:13
```

It'd be great if produced error could contain snippet of code pointing to problematic bit of code, similar to errors from TS and V8:

```ts
import b, { c } from "./b.js";

foo
console.log(b);
console.log(c);
```

```
error: Uncaught ReferenceError: foo is not defined
foo
^
    at file:///Users/biwanczuk/dev/deno/a.js:3:1
```

CC @kdy1

### LINKED ISSUE #15669: Deno formatter can break a switch's previously satisfied `no-fallthrough` linter rule
```
$ deno --version
> deno 1.24.3 (release, aarch64-apple-darwin)
> v8 10.4.132.20
> typescript 4.7.4
```

given this source, `deno lint --rules-include="no-fallthrough"` wants you to mark each line (or file) as fallthrough allowed:
```
const unit = 's';
let seconds = 10;

switch (unit) {
  case 'd': seconds *= 24;
  case 'h': seconds *= 60;
  case 'm': seconds *= 60;
}
```

adding this (instead of the longer `deno-lint-ignore no-fallthrough`) conveniently shuts it up:
```
const unit = 's';
let seconds = 10;

switch (unit) {
  case 'd': seconds *= 24; // fallthrough
  case 'h': seconds *= 60; // fallthrough
  case 'm': seconds *= 60; // fallthrough
}
```

but when you then run `deno fmt` the comments stay on the case's line:
```
const unit = "s";
let seconds = 10;

switch (unit) {
  case "d": // fallthrough
    seconds *= 24;
  case "h": // fallthrough
    seconds *= 60;
  case "m": // fallthrough
    seconds *= 60;
}
```

which in turn results in the `no-fallthrough` rule to fail (since it expects it to be on the last line before a new case starts)

### PR DESCRIPTION
**Pull request by dsherret**
Closes #17172
Closes #15669
Closes #8529

### CODE REVIEW

File: cli/standalone.rs [hunk 1]
```diff
@@ -74,9 +76,10 @@ pub async fn extract_standalone(
 ) -> Result<Option<(Metadata, eszip::EszipV2)>, AnyError> {
   let current_exe_path = current_exe()?;
 
-  let file = tokio::fs::File::open(current_exe_path).await?;
+  let file = std::fs::File::open(current_exe_path)?;
 
-  let mut bufreader = tokio::io::BufReader::new(file);
+  let mut bufreader =
+    deno_core::futures::io::BufReader::new(AllowStdIo::new(file));
```
Comment (bartlomieju) [inline cli/standalone.rs:82]: Why this change?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/standalone.rs (hunk 1 above)
Comment (dsherret) [reply cli/standalone.rs:82]: See https://github.com/denoland/eszip/pull/103

Comment (bartlomieju) [reply cli/standalone.rs:82]: Makes sense, thanks!

**Pull request merged**